ci/include-grouping.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+"""
+easy command line to run against all citus-style checked files:
+
+$ git ls-files \
+  | git check-attr --stdin citus-style \
+  | grep 'citus-style: set' \
+  | awk '{print $1}' \
+  | cut -d':' -f1 \
+  | xargs -n1 ./ci/include-grouping.py
+"""
+
+
+import os
+import sys
+
+
+def main(args):
+    if len(args) < 2:
+        print("Usage: include-grouping.py <file>")
+        return
+
+    file = args[1]
+    if not os.path.isfile(file):
+        print("File does not exist", file=sys.stderr)
+        return sys.exit(1)
+
+    with open(file, "r") as f:
+        with open(file + ".tmp", "w") as out_file:
+            lines = f.readlines()
+            includes = []
+            skipped_lines = []
+            for line in lines:
+                if line.startswith("#include"):
+                    includes.append(line)
+                    skipped_lines = []
+                else:
+                    if len(includes) > 0:
+                        # regroup all includes by skipping all empty lines while scanning
+                        if len(line.strip()) == 0:
+                            skipped_lines.append(line)
```

Comment:
If we move this line right after https://github.com/citusdata/citus/pull/7326/files#diff-4ef8be7a6b340207edfe9e3923061e33b918e3c0cb62dac9ec6989f882ddb716R37 that will allow to get rid of https://github.com/citusdata/citus/pull/7326/files#diff-4ef8be7a6b340207edfe9e3923061e33b918e3c0cb62dac9ec6989f882ddb716R51

Reply:
I had a bit of a hard time understanding these comments.

Nevertheless, the code in this area underwent significant rewriting.
Let me know if you still think this is applicable.